integration/utils/worker_stats_tracker.go

```diff
@@ -40,37 +41,38 @@ func (st *WorkerStatsTracker) AddWorker() {
 	st.mux.Lock()
 	defer st.mux.Unlock()
 
-	st.workers += 1
+	st.workers++
 }
 
 func (st *WorkerStatsTracker) AddTxSent() {
+	now := time.Now().Unix()
+
 	st.mux.Lock()
 	defer st.mux.Unlock()
 
-	st.txsSent += 1
-
-	now := time.Now().Unix()
-	if c, ok := st.txsSentPerSecond[now]; ok {
-		st.txsSentPerSecond[now] = c + 1
-		return
-	}
-	st.txsSentPerSecond[now] = 1
+	st.txsSent++
+	st.txsSentPerSecond[now]++
 }
 
 // AvgTPSBetween returns the average transactions per second TPS between the two time points
 func (st *WorkerStatsTracker) AvgTPSBetween(start, stop time.Time) float64 {
-	st.mux.Lock()
-	defer st.mux.Unlock()
-
 	sum := 0
+	toDelete := make([]int64, 0)
 
+	st.mux.Lock()
+	defer st.mux.Unlock()
 	for timestamp, count := range st.txsSentPerSecond {
-		if time.Unix(timestamp, 0).Before(start) {
+		if timestamp < start.Unix() {
+			toDelete = append(toDelete, timestamp)
 			continue
 		}
 		sum += count
 	}
 
+	for _, timestamp := range toDelete {
```

Comment:
seems we weren't deleting these before, i guess this was giving an incorrect average before then?

Reply:
It was mostly affecting the speed and memory usage, though given that there is only a single entry per second you need to run it for multiple days to notice any measurable effect.